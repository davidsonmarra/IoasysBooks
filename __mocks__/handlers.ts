import { rest } from 'msw';
import { mockFetchDataBooks, mockFetchDataRefresh, mockFetchDataSignIn } from '.';
import constants from '../src/constants';

export const handlers = [
  rest.get(`${constants.BASE_URL}/books`, (req, res, ctx) => {
    const search: string = req.url.searchParams.get('title') || '';
    return search.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataBooks));
  }),
  rest.post(`${constants.BASE_URL}/auth/sign-in`, async (req, res, ctx) => {
    const { email } = await req.json();
    return email.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataSignIn));
  }),
  rest.post(`${constants.BASE_URL}/auth/refresh-token`, async (req, res, ctx) => {
    const { refreshToken } = await req.json();
    return refreshToken.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataRefresh));
  })
];
