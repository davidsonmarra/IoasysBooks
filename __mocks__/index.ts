import BookDTO from '../src/@types/BookDTO';

interface IMockFetchDataBooks {
  data: BookDTO[];
  page: number;
  totalPages: number;
  totalItems: number;
}

export const mockBook: BookDTO = {
  id: '1',
  title: 'Test Title',
  imageUrl: 'https://www.test.com',
  authors: ['Test Author'],
  pageCount: 100,
  publisher: 'Test Publisher',
  published: 1980,
  category: 'Test Category',
  language: 'Test Language',
  isbn10: 'Test ISBN10',
  isbn13: 'Test ISBN13',
  description: 'Test Description'
};

export const mockPreloadedStateEmpty = {
  profile: {
    email: '',
    token: '',
    error: {} as Error,
    isLogged: false,
    isLoading: false
  },
  books: {
    isEnd: false,
    loadingFetchBooks: false,
    error: {} as Error,
    booksData: [],
    category: { key: '', title: '' },
    search: ''
  }
};

export const mockPreloadedStateFilled = {
  profile: {
    email: 'davidson',
    token: 'teste',
    error: {} as Error,
    isLogged: true,
    isLoading: false
  },
  books: {
    isEnd: false,
    loadingFetchBooks: false,
    error: {} as Error,
    booksData: [mockBook],
    category: { key: '', title: '' },
    search: ''
  }
};

export const mockFetchDataBooks: IMockFetchDataBooks = {
  data: [
    {
      ...mockBook,
      id: '2',
      title: 'Harry Potter 1'
    },
    {
      ...mockBook,
      id: '3',
      title: 'Harry Potter 2'
    },
    {
      ...mockBook,
      id: '4',
      title: 'Harry Potter 3'
    },
    {
      ...mockBook,
      id: '5',
      title: 'Harry Potter 4'
    },
    {
      ...mockBook,
      id: '6',
      title: 'Harry Potter 5'
    }
  ],
  page: 1,
  totalPages: 1,
  totalItems: 2
};
