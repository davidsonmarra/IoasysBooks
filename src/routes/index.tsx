import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { PublicRoutes } from './stacks/public.routes';
import { useSelector } from 'react-redux';
import { IRootState } from '../store/store';
import { AuthRoutes } from './stacks/auth.routes';

export function Routes() {
  const { isLogged } = useSelector(({ profile }: IRootState) => profile);

  return <NavigationContainer>{isLogged ? <AuthRoutes /> : <PublicRoutes />}</NavigationContainer>;
}
