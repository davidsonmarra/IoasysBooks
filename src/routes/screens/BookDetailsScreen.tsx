import React from 'react';
import { BookDetails } from '../../presentational';
import AuthRootStackParamList from '../../@types/AuthRootStackParamList';
import { RouteProp, useRoute } from '@react-navigation/native';

export function BookDetailsScreen() {
  const {
    params: { book }
  } = useRoute<RouteProp<AuthRootStackParamList, 'BookDetailsScreen'>>();

  return <BookDetails book={book} />;
}
