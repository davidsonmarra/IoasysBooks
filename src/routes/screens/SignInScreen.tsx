import React from 'react';
import { SignIn } from '../../presentational';

export function SignInScreen() {
  return <SignIn />;
}
