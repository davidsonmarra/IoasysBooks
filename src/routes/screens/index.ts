export { SignInScreen } from './SignInScreen';
export { HomeScreen } from './HomeScreen';
export { BookDetailsScreen } from './BookDetailsScreen';
