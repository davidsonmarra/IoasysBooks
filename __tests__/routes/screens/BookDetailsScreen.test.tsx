import { NavigationContainer } from '@react-navigation/native';
import { BookDetailsScreen } from '../../../src/routes/screens';
import { mockBook, mockPreloadedStateFilled } from '../../../__mocks__';
import { matchSnapshotWithProviders, renderWithCustomProviders } from '../../test-utils';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useRoute: () => ({
    params: {
      book: mockBook
    }
  })
}));

describe('Book Details Screen', () => {
  it('should match a snapshot', () => {
    matchSnapshotWithProviders(
      <NavigationContainer>
        <BookDetailsScreen />
      </NavigationContainer>
    );
  });

  it('should go back function be called correctly', () => {
    const {
      render: { getAllByText }
    } = renderWithCustomProviders(
      <NavigationContainer>
        <BookDetailsScreen />
      </NavigationContainer>,
      mockPreloadedStateFilled
    );
    expect(getAllByText(/test title/i)).toBeTruthy();
  });
});
