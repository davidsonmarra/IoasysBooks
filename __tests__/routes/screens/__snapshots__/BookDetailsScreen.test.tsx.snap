// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Details Screen should match a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E5E5E5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 31,
        "paddingLeft": 31,
        "paddingRight": 31,
        "paddingTop": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingTop": 88,
        },
      ]
    }
  >
    <View
      accessibilityLabel="go back"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#00000052",
          "borderRadius": 31,
          "borderStyle": "solid",
          "borderWidth": 2,
          "height": 63,
          "justifyContent": "center",
          "opacity": 1,
          "width": 63,
        }
      }
    >
      <SvgMock
        style={
          Array [
            Object {},
          ]
        }
      />
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 31,
      }
    }
    showsVerticalScrollIndicator={false}
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 47,
          "paddingRight": 47,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://www.test.com",
          }
        }
        style={
          Array [
            Object {
              "alignSelf": "center",
              "height": 687,
              "marginTop": 39,
              "shadowColor": "rgba(84, 16, 95, 0.13)",
              "shadowOffset": Object {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 24,
              "width": 471,
            },
          ]
        }
      />
      <Text
        numberOfLines={2}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#333333",
              "fontFamily": "Heebo-Medium",
              "fontSize": 55,
              "marginTop": 47,
              "textAlign": "center",
            },
          ]
        }
      >
        Test Title
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
              "marginBottom": 31,
            },
          ]
        }
      >
        <Text
          isLast={true}
          style={
            Array [
              Object {
                "color": "#B22E6F",
                "fontFamily": "Heebo-Regular",
                "fontSize": 24,
                "marginRight": 8,
              },
            ]
          }
        >
          Test Author 1
          ,
        </Text>
        <Text
          isLast={false}
          style={
            Array [
              Object {
                "color": "#B22E6F",
                "fontFamily": "Heebo-Regular",
                "fontSize": 24,
                "marginRight": 0,
              },
            ]
          }
        >
          Test Author 2
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Heebo-Medium",
              "fontSize": 24,
              "marginBottom": 31,
              "marginTop": 31,
              "textTransform": "uppercase",
            },
          ]
        }
      >
        Informações
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 24,
              },
            ]
          }
        >
          Páginas
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Heebo-Regular",
                "fontSize": 24,
              },
            ]
          }
        >
          100
           páginas
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 24,
              },
            ]
          }
        >
          Editora
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Heebo-Regular",
                "fontSize": 24,
              },
            ]
          }
        >
          Test Publisher
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 24,
              },
            ]
          }
        >
          Publicação
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Heebo-Regular",
                "fontSize": 24,
              },
            ]
          }
        >
          1980
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 24,
              },
            ]
          }
        >
          Idioma
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Heebo-Regular",
                "fontSize": 24,
              },
            ]
          }
        >
          Test Language
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 24,
              },
            ]
          }
        >
          Título Original
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Heebo-Regular",
                "fontSize": 24,
              },
            ]
          }
        >
          Test Title
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 24,
              },
            ]
          }
        >
          ISBN-10
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Heebo-Regular",
                "fontSize": 24,
              },
            ]
          }
        >
          Test ISBN10
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 24,
              },
            ]
          }
        >
          ISBN-13
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Heebo-Regular",
                "fontSize": 24,
              },
            ]
          }
        >
          Test ISBN13
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 24,
              },
            ]
          }
        >
          Categoria
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "Heebo-Regular",
                "fontSize": 24,
              },
            ]
          }
        >
          Test Category
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Heebo-Medium",
              "fontSize": 24,
              "marginBottom": 31,
              "marginTop": 31,
              "textTransform": "uppercase",
            },
          ]
        }
      >
        Resenha da Editora
      </Text>
      <SvgMock
        style={
          Array [
            Object {
              "marginBottom": 16,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#999999",
              "fontFamily": "Heebo-Regular",
              "fontSize": 24,
            },
          ]
        }
      >
        Test Description
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
